### PR TITLE
[SuiteSparse] Have SuiteSparse depend on METIS_jll

### PR DESCRIPTION
**Pull request by ViralBShah**
Based on discussions in #289, make METIS_jll a dependency for SuiteSparse. Should be retriggered after #290 is merged.

### CODE REVIEW

File: S/SuiteSparse@5/build_tarballs.jl [hunk 1]
```diff
@@ -78,9 +60,9 @@ if [[ ${target} == *-apple-* ]] || [[ ${target} == *freebsd* ]]; then
     done
 fi
 
-# Compile suitesparse_wrapper shim
+# Compile SuiteSparse_wrapper shim
 cd $WORKSPACE/srcdir/SuiteSparse_wrapper
-make "${FLAGS[@]}" install
+"${CC}" "${CFLAGS}" -O2 -shared -fPIC -I${prefix}/include SuiteSparse_wrapper.c -o ${libdir}/libsuitesparse_wrapper.${dlext} "{LDFLAGS}" -L${libdir} -lcholmod -lumfpack -lspqr
```
Comment (giordano) [inline S/SuiteSparse@5/build_tarballs.jl]: Good catch!

File: S/SuiteSparse@5/build_tarballs.jl [hunk 2]
```diff
@@ -17,7 +17,7 @@ cd $WORKSPACE/srcdir/SuiteSparse/
 # Apply Jameson's shlib patch
 atomic_patch -p1 ${WORKSPACE}/srcdir/patches/SuiteSparse-shlib.patch
 # Disable OpenMP as it will probably interfere with blas threads and Julia threads
-FLAGS=(INSTALL="${prefix}" INSTALL_LIB="${libdir}" INSTALL_INCLUDE="${prefix}/include" CFOPENMP=)
+FLAGS=(INSTALL="${prefix}" INSTALL_LIB="${libdir}" INSTALL_INCLUDE="${prefix}/include" MY_METIS_LIB="-lmetis" CFOPENMP=)
```
Comment (giordano) [inline S/SuiteSparse@5/build_tarballs.jl]: Is `MY_METIS_LIB` used anywhere?

Comment (giordano) [reply S/SuiteSparse@5/build_tarballs.jl]: Oh, it's in the Makefiles.

Comment (ViralBShah) [reply S/SuiteSparse@5/build_tarballs.jl]: SuiteSparse_config should pick it up.

https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/79f25b523ae0fd81abe1ffb0efd9005f2e6eef33/SuiteSparse_config/SuiteSparse_config.mk#L491

**Comment by ViralBShah** [comment]
Is there a way to re-trigger the build? I had to merge METIS_jll in the registry.

**Comment by ViralBShah** [comment]
The failure is because of the inability to find metis.h. Does METIS_jll install it in `${prefix}/include`?

File: S/SuiteSparse@5/build_tarballs.jl (hunk 1 above)
Comment (ViralBShah) [reply S/SuiteSparse@5/build_tarballs.jl]: I think it worked fine, but the handcrafted Makefile was best avoided.

**Comment by giordano** [comment]
Note: before merging this I think we should close https://github.com/JuliaRegistries/General/pull/6701 and delete the [`SuiteSparse-v5.4.0+2`](https://github.com/JuliaBinaryWrappers/SuiteSparse_jll.jl/releases/tag/SuiteSparse-v5.4.0%2B2) tag of `SuiteSparse_jll.jl`.

**Comment by giordano** [comment]
No, wait!  Don't delete the tag and don't merge.

**Comment by giordano** [comment]
I'm completely lost.  Merge of #289 didn't trigger a new registration in the General registry.  I fear this is the cause of https://github.com/JuliaLang/julia/pull/34082#issuecomment-565837625

**Comment by ViralBShah** [comment]
Ok, Not sure I am fully following, but won't hit merge. It is pretty much ready to go whenever things are in place.

**Comment by ViralBShah** [comment]
I didn't realize that publishing new binaries here will mean that Julia will start using the new ones. I thought that we'd have to bump them for that to happen.

**Comment by giordano** [comment]
I wrote down what went wrong in https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/554

**Comment by ViralBShah** [comment]
If Julia is going to use these new binaries, we will need to make it pull in METIS as well.

**Comment by ViralBShah** [comment]
What's the right series of operations to do next? @giordano Can I request you to do the merges in the right order?

**Comment by giordano** [comment]
My roadmap is:
* merge this
  * the artifacts in [`SuiteSparse-v5.4.0+2`](https://github.com/JuliaBinaryWrappers/SuiteSparse_jll.jl/releases/tag/SuiteSparse-v5.4.0%2B2) will automatically be updated
  * https://github.com/JuliaRegistries/General/pull/6701 will automatically be updated
* merge https://github.com/JuliaRegistries/General/pull/6701
* bring back https://github.com/JuliaLang/julia/pull/34082 but fixing the hashes of SuiteSparse
However I'd wait for @staticfloat to decide

**Comment by ViralBShah** [comment]
Sorry for the inconvenience here. I didn't realize that this had to be carefully orchestrated here.

**Comment by giordano** [comment]
No worries.  I think what happened has been an incredible unlikely coincidence of a few events.

**Comment by vchuravy** [comment]
Can we have the original version? E.g. revert #289 and then do #289 and #291 as SuiteSparse-v5.4.0+3 ?

**Comment by giordano** [comment]
Do you want the version without metis in Julia?

**Comment by vchuravy** [comment]
I want to minimize the amount of potential breakage, we are at the tail end of the development window for Julia 1.4 and the version without Metis went through CI, so I would prefer to be conservative and figure out how to handle the Metis dependency properly in the next cycle.

**Comment by giordano** [comment]
Makes sense to me

**Comment by giordano** [comment]
Ok, https://github.com/JuliaRegistries/General/pull/6701 has been merged with https://github.com/JuliaBinaryWrappers/SuiteSparse_jll.jl/releases/tag/SuiteSparse-v5.4.0%2B2 providing the artifacts without metis.   It looks like this can be merged straight away without having to re-do #289 or rebase this PR.

**Comment by ViralBShah** [comment]
Just making sure - this is good to merge? It is good to go from my side.

**Comment by giordano** [comment]
Yes, it should be safe to merge now, version 5.4.0+2 of SuiteSparse_jll is in [the General registry](https://github.com/JuliaRegistries/General/blob/1031b0af7f8bb20e0e0c0d8e96fd1672772305d3/S/SuiteSparse_jll/Versions.toml)

**Comment by ViralBShah** [comment]
Do I need to merge something in the General registry next?

**Comment by giordano** [comment]
This is interesting: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=594&view=logs&j=12f17990-53ef-5136-c697-70a8cd1d430e&t=47ad8431-de36-5e15-22e3-706b0aec61ee.  Registration failed
```
┌ Error: Error in (Julia)Project.toml: Package 'METIS_jll' with UUID: d00139f3-1899-568f-a2f0-47f597d42d70 not found in registry or stdlib
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/2qObQ/src/AutoBuild.jl:445
```
but apparently it didn't return an exit code different from 0, so the job seems to be successful.  @staticfloat syncing problems?

**Pull request merged**